test-suite/js/src/main/scala/org/scalajs/testsuite/utils/Platform.scala

```diff
@@ -32,20 +33,37 @@ object Platform {
   }
 
   def jsSymbols: Boolean =
-    assumeES2015 || js.typeOf(js.Dynamic.global.Symbol) != "undefined"
+    assumedESVersion >= ESVersion.ES2015 || js.typeOf(js.Dynamic.global.Symbol) != "undefined"
 
   def typedArrays: Boolean =
-    assumeES2015 || js.typeOf(js.Dynamic.global.Int32Array) != "undefined"
+    assumedESVersion >= ESVersion.ES2015 || js.typeOf(js.Dynamic.global.Int32Array) != "undefined"
 
   def jsMaps: Boolean =
-    assumeES2015 || js.typeOf(js.Dynamic.global.Map) != "undefined"
+    assumedESVersion >= ESVersion.ES2015 || js.typeOf(js.Dynamic.global.Map) != "undefined"
 
   def jsBigInts: Boolean =
-    js.typeOf(js.Dynamic.global.BigInt) != "undefined"
+    assumedESVersion >= ESVersion.ES2020 || js.typeOf(js.Dynamic.global.BigInt) != "undefined"
+
+  lazy val jsRegExps2018: Boolean =
+    assumedESVersion >= ESVersion.ES2018 || regexFeatureTest("(?<=a)(?<!b)\\p{L}\\P{L}", "us")
 
   def sourceMaps: Boolean = BuildInfo.hasSourceMaps && executingInNodeJS
 
-  def assumeES2015: Boolean = BuildInfo.es2015
+  /** The assumed ECMAScript version.
+   *
+   *  Prefer this property over `assumeES2015` when testing for support of
+   *  library functions.
```

Comment:
I'm not sure this advice actually applies here: It seems the pattern emerges to do JS library feature tests in this file. So maybe we should say don't use this (except in this file and for the `LinkingInfoTest`)?

Reply:
Some library feature tests that are repeatedly used in the codebase are done here. But there are a few additional ones that are done on a case-by-case basis. For examples, the tests in `ObjectTest`.